pyrsistent/_checked_types.py

```diff
@@ -344,8 +344,12 @@ def _check(self, it):
             self._invariant_errors.extend(error_data)
 
         def add(self, element):
-            self._check([element])
-            self._pmap_evolver[element] = True
+            return self.update([element])
+
+        def update(self, elements):
+            elements = frozenset(elements)
+            self._check(elements)
+            self._pmap_evolver.update(dict(zip(elements, repeat(True))))
```

Comment:
That's a common trick to implement a set if you have an implementation of a map. A set is just a map where the keys are the items in the set and the value is always simply True if the object is in the set.

It appears pyrsistent has chosen this implementation strategy for code re-use between set and map.